### PR TITLE
updates

### PR DESCRIPTION
**Pull request by Kyleeld**
high alch profit highlighter

moveable multi combat widget

pvp prayer rework

individual prayer rework

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	private int getHAProfit(int haPrice, int gePrice, int alchCost)
+	{
+		return haPrice - gePrice - alchCost;
+	}
+
+	private int getAlchCost()
+	{
+		// int natureRunePrice = itemManager.getItemPrice(ItemID.NATURE_RUNE);
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: Forgot to uncomment?

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		int natureRunePrice = 0;
+		if (config.usingFireRunes())
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: I think it'd be better to check if the player is wearing tome of fire/a staff providing infinite fire runes

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		{
+			int fireRunePrice = itemManager.getItemPrice(ItemID.FIRE_RUNE) * 5;
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: Lava/steam/smoke runes? I personally never use fire runes as lavas are cheaper, may be better to check inventory/rune pouch for that

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	//Checks if item is noted, if not returns id
+	private int getNotedId(int id)
+	{
+		int noteID = id;
+		ItemComposition itemComposition = itemManager.getItemComposition(noteID);
+		if (itemComposition.getNote() != -1)
+		{
+			noteID = itemComposition.getLinkedNoteId();
+		}
+		return noteID;
```
Comment (Lucwousin) [inline runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: Shouldn't the logic for this + price be handled in the plugin?

Also, you're currently calculating the prices and such on every frame. Isn't there a way to only calculate the price when the item changes?

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java (hunk 2 above)
Comment (Kyleeld) [reply runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: just leave the config blank if youre not using a fire rune, the options there for if youre not using a tome/staff

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java (hunk 3 above)
Comment (Kyleeld) [reply runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: Only really useful for slayer, which you dont care about profit as youll alch drops anyway from things such as gargoyles

File: runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java (hunk 1 above)
Comment (Kyleeld) [reply runelite-client/src/main/java/net/runelite/client/plugins/highalchemy/HighAlchemyOverlay.java]: yup, updated

**Pull request merged**